lib/src/local_working_copy.rs

```diff
@@ -1041,6 +1051,18 @@ impl TreeState {
             }
             deleted_files
         });
+        trace_span!("write tree").in_scope(|| {
+            let new_tree_id = tree_builder.write_tree(&self.store).unwrap();
+            if dry_run {
+                stats.dry_run_tree_id = Some(new_tree_id);
```

Comment:
nit: It seems odd that we have `dry_run_tree_id` which is conditionally set. It might be better to add `LockedWorkignCopy::snapshot_dry_run() -> (MergedTreeId, SnapshotStats)`. It might also make sense to add the function to `WorkingCopy` because dry-run `snapshot()` wouldn't need locking.